heartbeat/docs/reference/configuration/heartbeat-options.asciidoc

```diff
@@ -0,0 +1,448 @@
+
+[[monitor-tcp-tls-ssl]]
+===== ssl
+
+The TLS/SSL connection settings. 
+
+//Steffen: Is this option so that you can check to see if the servers certificates are valid?
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: tcp 
+  schedule: '@every 5s'
+  hosts: ["myhost"]
+  ports: [80, 9200, 5044]
+  ipv4: true
+  ipv6: true
+  mode: any
+  timeout: 16s
+  ssl:
+    certificate_authorities: ['/etc/ca.crt']
+    supported_protocols: ["TLSv1.0", "TLSv1.1", "TLSv1.2"]
+-------------------------------------------------------------------------------
+
+
+//Steffen: How are the values for supported_protocols used?
+
+[[monitor-http-options]]
+==== HTTP Options
+
+These options are valid when the `type` is `http`.
+
+[[monitor-http-urls]]
+===== urls
+
+A list of URLs to ping.
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: http 
+  schedule: '@every 5s'
+  urls: ["http://myhost:80"]
+  ipv4: true
+  ipv6: true
+  mode: any
+-------------------------------------------------------------------------------
+
+
+[[monitor-http-proxy-url]]
+===== proxy_url
+
+The HTTP proxy URL. This setting is optional. If not set, the HTTP_PROXY
+environment variable is used.
+
+[[monitor-http-username]]
+===== username
+
+The username for authenticating with the server. This setting is optional. 
+
+[[monitor-http-password]]
+===== password
+
+The password for authenticating with the server. This setting is optional.
+
+//Steffen: When are authentication credential required? If you are sending a request and expecting a response?
+
+[[monitor-http-tls-ssl]]
+===== ssl
+
+The TLS/SSL connection settings for use with the HTTPS endpoint. If you don't
+specify settings, the system defaults are used. 
+
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: http 
+  schedule: '@every 5s'
+  urls: ["http://myhost:80"]
+  ipv4: true
+  ipv6: true
+  mode: any
+  ssl:
+    certificate_authorities: ['/etc/ca.crt']
+    supported_protocols: ["TLSv1.0", "TLSv1.1", "TLSv1.2"]  
+-------------------------------------------------------------------------------
+
+
+[[monitor-http-check]]
+===== check
+
+An optional `request` to send to the remote host and the expected `response`. 
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: http 
+  schedule: '@every 5s'
+  urls: ["http://myhost:80"]
+  ipv4: true
+  ipv6: true
+  mode: any
+  check.request:
+    method: "GET"
+    headers:
+    body:
+  check.response:
+    status: 0
+    headers:
+    body:
+-------------------------------------------------------------------------------
+
+
+//Steffen: I need help filling out the above example with headers and body
```

Comment:
how about something very simple:

```
check.request.method: HEAD
check.response.status: 200
```

These checks are very specific to a service.

e.g. some check with JSON response:

```
check.request:
  method: GET
  headers:
    'X-API-Key': '12345-mykey-67890'
check.response:
  status: 200
  body: '{"status": "ok"}'
```